2020/day3.clj

```diff
@@ -7,27 +7,34 @@
        (string/split-lines)))
 
 (defn next-location
-  [loc [right down]]
-  [(+ (first loc) right)
-   (+ (second loc) down)])
+  [[row col] [right down]]
+  [(+ row right)
+   (+ col down)])
 
 (defn locations-s
-  [height [right down]]
-  (take-while #(< (second %) height)
-              (map #(next-location [0 0] [(* right %) (* down %)])
-                   (range))))
+  [slope]
+  (map (fn [n] (next-location [0 0] [(* (first slope) n) (* (second slope) n)]))
+       (range)))
+
+(defn valid-locations
+  [height locations]
+  (->> locations
+       (take-while (fn [loc] (< (second loc) height)))))
+
+(defn char-at-location
+  [grid loc]
+  (get-in grid [(second loc) (mod (first loc) (count (first grid)))]))
 
 (defn tree?
   [ch]
   (= ch \#))
 
 (defn count-trees-on-slope
   [grid [right down]]
-  (let [height          (count grid)
-        valid-locations (locations-s height [right down])]
+  (let [locations       (locations-s [right down])
+        valid-locations (valid-locations (count grid) locations)]
```

Comment:
(count grid) doesn't mean anything on it's own...

Reply:
Also, there's no reason to cut this off here

Reply:
> (count grid) doesn't mean anything on it's own...

using height for "(count grid)".

Reply:
> Also, there's no reason to cut this off here, this seq should remain infinite. You only want to cut this off AFTER you combine it with the grid

I've rewrote this code. I cut this off in the threading and sequence of locations remains infinite until combined with the grid.